Core/SiteRating.swift

```diff
@@ -71,3 +71,30 @@ public struct SiteRating {
         return SiteGrade.grade(fromScore: siteScore)
     }
 }
+
+
+public class SiteRatingCache {
+    
+    public static let shared = SiteRatingCache()
+    
+    private var cachedScores = [String: Int]()
+
+    
+    /**
+     Registers a site score for in-memory caching. Only updates the score
+     in the cache if the new score is higher.
+     - returns: The score maintaned in the cache
+     */
+    func register(domain: String, score current: Int) -> Int {
```

Comment:
I've gone with the former, let me know if that reads better now. If so, I'll squash the changes (rather than merge) them in